rfcs/text/remove-support-for-uncontrolled-inputs.md

```diff
@@ -0,0 +1,234 @@
+  defaultValue: PropTypes.string,
+};
+
+export default MyComponent;
+```
+Here is an example of how a basic component will look in TS with uncontrolled support.
+
+```tsx
+// my-component.component.tsx
+interface MyComponentProps = {
+  value?: string;
+  onChange?: (event: React.ChangeEvent<HTMLInputElement>) => void;
+  defaultValue?: string;
+};
+
+const MyComponent = ({ value, onChange, defaultValue, ...rest }: MyComponentProps) => {
+  const isControlled = useRef(value !== undefined);
+  const inputRef = useRef(null);
+
+   const handleOnChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+    if (isControlled && onChange ) {
+      onChange(event);
+    } 
+  };
+
+  useEffect(() => {
+    // invariant goes in here.
+  }, [];
+
+  /* Component logic goes here */
+
+return (
+    <input 
+      value={value} 
+      onChange={handleOnChange} 
+      defaultValue={defaultValue} 
+      ref={inputRef}
+    />
+  )
+};
+
+export default MyComponent;
+```
+## Component Example - Controlled
+
+Here is an example of how a basic component will look in JS without uncontrolled support.
+
+```jsx 
+// my-component.component.js
+const MyComponent = ({ value, onChange, ...rest }) => {
+   const handleOnChange = (event) => {
+    if (onChange) {
+      onChange(event);
+    } 
+  };
+
+  /* Component logic goes here */
+
+  return <input value={value} onChange={handleOnChange} />;
+};
+
+MyComponent.propTypes = {
+  value: PropTypes.string.isRequired,
+  onChange: PropTypes.function.isRequired,
+};
+
+export default MyComponent;
+```
+Here is an example of how a basic component will look in TS without uncontrolled support.
+
+```tsx
+// my-component.component.tsx
+interface MyComponentProps = {
+  value: string;
+  onChange: (event: React.ChangeEvent<HTMLInputElement>) => void;
+};
+
+const MyComponent = ({ value, onChange, ...rest }: MyComponentProps) => {
+   const handleOnChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+    if (onChange) {
+      onChange(event.target.value);
+    } 
+  };
+
+  /* Component logic goes here */
+
+  return <input value={value} onChange={handleOnChange} />;
+};
+
+export default MyComponent;
+```
+# Detailed design
+
+- We will add deprecation warnings to our input components so that our consumers are aware of the pending changes to remove uncontrolled support. This will appear as a console warning if our consumer is using the uncontrolled mode and a warning on our Storybook documentation.
+
+- We will remove uncontrolled behaviour from our input components and associated props. This will be done in one Pull Request with each component having it's own commit message with a `BREAKING CHANGE`. The benefit of this is that we will only release one major version of Carbon. 
+
+This approach will provide immediate benefit to developers of Carbon as it means we will have less code and tests to maintain. It will also mean that consumers of Carbon can steadily upgrade their version of Carbon.
```

Comment:
```suggestion
```

We're doing one release so consumers will have to take a big change.

Reply:
Removed this point.